src/GovUk.Education.ExploreEducationStatistics.Content.Api/Controllers/ThemeController.cs

```diff
@@ -14,50 +14,58 @@ namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Controllers
+        private readonly IFileStorageService _fileStorageService;
 
-        public ThemeController(IBlobStorageService blobStorageService)
+        public ThemeController(IFileStorageService fileStorageService)
         {
-            _blobStorageService = blobStorageService;
+            _fileStorageService = fileStorageService;
         }
 
         [HttpGet("themes")]
         public async Task<ActionResult<IEnumerable<ThemeTree<PublicationTreeNode>>>> GetThemes()
         {
-            return await this.JsonContentResultAsync<IEnumerable<ThemeTree<PublicationTreeNode>>>(
-                () =>
-                    _blobStorageService.DownloadBlobText(
-                        PublicContentContainerName,
-                        PublicContentPublicationsTreePath()
-                    ),
-                NoContent()
-            );
+            try
+            {
+                return await _fileStorageService
+                    .GetDeserialized<IEnumerable<ThemeTree<PublicationTreeNode>>>(PublicContentPublicationsTreePath())
+                    .HandleFailuresOrOk();
+            }
+            catch (JsonException)
+            {
+                return new NoContentResult();
+            }
         }
 
         [HttpGet("download-themes")]
         public async Task<ActionResult<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>> GetDownloadThemes()
         {
-            return await this.JsonContentResultAsync<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>(
-                () =>
-                    _blobStorageService.DownloadBlobText(
-                        PublicContentContainerName,
+            try
+            {
+                return await _fileStorageService
+                    .GetDeserialized<IEnumerable<ThemeTree<PublicationDownloadTreeNode>>>(
                         PublicContentDownloadTreePath()
-                    ),
-                NoContent()
-            );
+                    )
+                    .HandleFailuresOrOk();
+            }
+            catch (JsonException)
+            {
+                return new NoContentResult();
+            }
         }
 
         [HttpGet("methodology-themes")]
         public async Task<ActionResult<IEnumerable<ThemeTree<MethodologyTreeNode>>>> GetMethodologyThemes()
         {
-            return await this.JsonContentResultAsync<IEnumerable<ThemeTree<MethodologyTreeNode>>>(
-                () =>
-                    _blobStorageService.DownloadBlobText(
-                        PublicContentContainerName,
-                        PublicContentMethodologyTreePath()
-                    ),
-                NoContent()
-            );
+            try
+            {
+                return await _fileStorageService
+                    .GetDeserialized<IEnumerable<ThemeTree<MethodologyTreeNode>>>(PublicContentMethodologyTreePath())
+                    .HandleFailuresOrOk();
+            }
+            catch (JsonException)
```

Comment:
Same as above, not sure why we turn this into a `NoContentResult`, although I suspect it's related to leaking implementation details like we talked about.

Reply:
Same reasons as above. All these endpoints will now throw 500s if there's a problem.